cmd/ctl/pkg/install/uninstall.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package install
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/release"
+
+	helm "github.com/jetstack/cert-manager/cmd/ctl/pkg/install/helm"
+)
+
+const uninstallDesc = `
+This command uninstalls cert-manager.
+
+It can uninstall cert-manager even if it was installed by another install tool.
+
+This command will also delete CRD resources when providing the '--remove-crds' flag.
+It is safer to use this cli than using helm directly (which might automatically remove
+cert-manager crds when uninstalling, depending on the install parameters).
+
+The tool first tries to find a helm-based cert-manager install (installed directly by helm
+or by this cli tool) and removes the resources based on the found helm release. In case no
+helm-based cert-manager install is found, a kubernetes manifest yamls are generated from
+the provided chart and chart parameters and are used to determine what resources to remove
+from the kubernetes cluster.
```

Comment:
I don't like this.
I'd prefer `uninstall` bail out unless it detects a single cert-manager installed by `kubectl cert-manager install`.

Generating static manifests from the latest Helm chart and assuming that those are the resources that are currently installed is unreliable. 
What if the latest release has removed various  RBAC cluster-role-bindings.
Or maybe we'll end up removing the conversion webhook once we remove the old API versions.
This strategy will result in those old resources being orphaned.

I think we need to consider this more carefully.

Reply:
Fixed it